content/advent-2017/custom-JSON-unmarshaler-for-GraphQL-client.md

```diff
@@ -0,0 +1,558 @@
++++
+author = ["Dmitri Shuralyov"]
+date = "2017-12-08T09:15:00Z"
+title = "Custom JSON unmarshaler for a GraphQL client"
+series = ["Advent 2017"]
++++
+
+In this post, I will tell a story of how I had to build a custom JSON unmarshaler for the needs of a GraphQL client library in Go. I'll start with the history of how everything started, build motivation for why a custom JSON marshaler was truly needed, and then describe how it was implemented. This is going to be a long journey, so strap yourself in, and here we go!
```

Comment:
Opening might be stronger if you focus more on the surprising aspects of the problem and less on your own journey to get there.

Reply:
Point taken.

But, I'm not sure there was much surprising stuff (aside from the dual slices of values maybe), it was a pretty straightforward problem (perhaps in hindsight?). I like that the current description is very accurate and describes exactly what the post talks about. I hope it attracts people who want to learn about what it takes to unmarshal JSON and/or how GraphQL works, as boring as that might be.

In other words, I think this post largely documents the normal course of solving a problem, and the description here backs that up. I can't think of any important surprising aspects.